service/devdoc/twin_requirements.md

```diff
@@ -103,28 +103,44 @@ having to get it first which can be useful in some high performance scenarios.
+  - `this.deviceId`
+  - `this.moduleId`
+  - `done`
+**]**
+
 ### update(patch, done)
-The `update` method is used to update any part of the device twin stored in the IoT Hub registry with the values passed as a patch argument. 
+The `update` method is used to update any part of the device twin stored in the IoT Hub registry with the values passed as a patch argument.
 To satisfy optimistic concurrency requirements, the HTTP request `If-Match` header will be populated with the `Twin.eTag` field if defined. If not defined, `*` will be used.
 
-**SRS_NODE_IOTHUB_TWIN_16_019: [** The `update` method shall call the `updateTwin` method of the `Registry` instance stored in `_registry` property with the following parameters:
+**SRS_NODE_IOTHUB_TWIN_16_019: [** If `this.moduleId` is `null`, The `update` method shall call the `updateTwin` method of the `Registry` instance stored in `_registry` property with the following parameters:
 - `this.deviceId`
 - `patch`
 - `this.etag`
 - `done`
 **]**
 
+**SRS_NODE_IOTHUB_TWIN_18_002: [** If `this.moduleId` is not `null`, the `update` method shall call the `updateModuleTwin` method of the `Registry` instance stored in `_registry` property with the following parameters:
+  - `this.deviceId`
+  - `this.moduleId`
+  - `patch`
+  - `this.etag`
+  - `done`
+**]**
+
 ### All twin methods calling to a Registry API
-**SRS_NODE_IOTHUB_TWIN_16_021: [** The method shall copy properties, tags, and etag in the twin returned in the callback of the `Registry` method call into its parent object. **]**  
-**SRS_NODE_IOTHUB_TWIN_16_022: [** The method shall call the `done` callback with an `Error` object if the request failed **]**  
-**SRS_NODE_IOTHUB_TWIN_16_023: [** The method shall call the `done` callback with a `null` error object, its parent instance as a second argument and the transport `response` object as a third argument if the request succeeded. **]**  
+**SRS_NODE_IOTHUB_TWIN_16_021: [** The method shall copy properties, tags, and etag in the twin returned in the callback of the `Registry` method call into its parent object. **]**
+**SRS_NODE_IOTHUB_TWIN_16_022: [** The method shall call the `done` callback with an `Error` object if the request failed **]**
+**SRS_NODE_IOTHUB_TWIN_16_023: [** The method shall call the `done` callback with a `null` error object, its parent instance as a second argument and the transport `response` object as a third argument if the request succeeded. **]**
 
 ### toJSON()
 The `toJSON` method is called when calling `JSON.stringify()` on a `Twin` object.
 
-**SRS_NODE_IOTHUB_TWIN_16_015: [** The `toJSON` method shall return a copy of the `Twin` object that doesn't contain the `_registry` private property. **]**
\ No newline at end of file
+**SRS_NODE_IOTHUB_TWIN_16_015: [** The `toJSON` method shall return a copy of the `Twin` object that doesn't contain the `_registry` private property. **]**
+
+
```

Comment:
Blank line blank line!!!!!

Reply:
I am so glad that you're here!

---
In reply to: [176865692](https://github.com/Azure/azure-iot-sdk-node/pull/256#discussion_r176865692) [](ancestors = 176865692)